code/controllers/subsystem/redis.dm

```diff
@@ -0,0 +1,99 @@
+SUBSYSTEM_DEF(redis)
+	name = "Redis"
+	init_order = INIT_ORDER_REDIS
+	runlevels = RUNLEVELS_DEFAULT | RUNLEVEL_LOBBY // ALL THE THINGS
+	wait = 1
+	flags = SS_TICKER // Every tick
+	/// Are we connected
+	var/connected = FALSE
+	/// Amount of subscribed channels on the redis server
+	var/list/subbed_channels = list()
+	/// Message queue (If messages are sent before the SS has init'd)
+	var/list/datum/redis_message/queue = list()
+	// remove this when the PR is deemed stable
+	var/last_send = 0
+
+// Redis operations
+// These will be moved in the final PR
+#define rustg_redis_connect(connstring) call(RUST_G, "redis_connect")(connstring)
+#define rustg_redis_disconnect call(RUST_G, "redis_disconnect")()
+#define rustg_redis_subscribe(channel) call(RUST_G, "redis_subscribe")(channel)
+#define rustg_redis_get_messages call(RUST_G, "redis_get_messages")()
+#define rustg_redis_publish(channel, message) call(RUST_G, "redis_publish")(channel, message)
+#define RUST_REDIS_ERROR_CHANNEL "RUSTG_REDIS_ERROR_CHANNEL"
+
+/datum/controller/subsystem/redis/stat_entry()
+	..("S:[length(subbed_channels)] | Q:[length(queue)]")
+
+/datum/controller/subsystem/redis/Initialize()
+	if(world.system_type == UNIX)
+		return ..() // Hack to bypass CI in debug mode
+	// Connect to cappuccino
+	connect()
+	if(connected)
+		// Loop efficiency doesnt matter here. It runs once and likely wont have any events in
+		for(var/datum/redis_message/RM in queue)
+			publish(RM.channel, RM.message)
+
+	// Subscribe to the debug send
+	subscribe("aa07.out.debug")
```

Comment:
Yes, code review.

I know the PR is full of hardcoded stuff. I am not doing config changes till I can prove the stability on this.

Reply:
^ This also applies for making a RUSTG release